### ISSUE TITLE
Views should enforce schemas

### ISSUE THREAD

**Issue by serpens24** [constellation-app/constellation#339]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Running the latest version of Constellation
	* Attached the ***Support Package*** via `Help` > `Support Package`
    * Checked the FAQs: https://github.com/constellation-app/constellation/wiki/FAQ
    * Checked that your issue isn't already filed: https://github.com/constellation-app/constellation/issues
    * Checked that there is not already a module that provides the described functionality: https://github.com/constellation-app/constellation/wiki/Catalogue-of-Repositories

### Description

Constellation has long had an issue where a plugin could create attributes on a graph whose schema does not support them, and the behaviour has always been that that plugin should fail.
I think instead we should allow plugins to create any attributes they want, and have views enforce schemas. For example, you might have a plugin create latitude and longitude attributes on a graph, but unless that graph is backed by the Analytic Schema (or one of its children) the Map View will not function as that is the minimum requirement of that view.
We could build this functionality into the view framework - perhaps if a view is opened on top of an incompatible graph, it will default to a pane informing the user that their graph is incompatible with this view.

### Steps to Reproduce

N/A

**Expected behaviour:**

Plugins can create attributes that are incompatible with underlying schemas.

**Actual behaviour:** 

As per description, investiagte the statement: "We could build this functionality into the view framework - perhaps if a view is opened on top of an incompatible graph, it will default to a pane informing the user that their graph is incompatible with this view." and implement if possible.

**Reproduces how often:** 

Unable to quantify.

### Additional Information

N/A

**Comment by aldebaran30701** [comment]
Is the extent of this ticket to create a popup when the view is not compatible with the schema?

**Comment by arcturus2** [comment]
@aldebaran30701 yes but rather than a popup, some sort of overlay over the view with a small message saying something like "Geo.Latitude and Geo.Longitute where not found on your graph". The view should be disabled so that there is no doubts as to why nothing is showing.

**Comment by aldebaran30701** [comment]
To follow up: different graph schemas will not work on all views. Simple Schema will not have default attributes content, geo or date time. When these schemas are in use, or if the current graph does not have the correct attributes registered that the view requires - The view should be disabled when opened. Upon adding those attributes the view should become active.
Constellation should do nothing behind the users view - Constellation should alert the user that there is not a prerequisite met and therefore prompt them to add it if needed.
